providers/slack/resources/users.go

```diff
@@ -0,0 +1,259 @@
+		ApiAppID:              u.ApiAppID,
+		StatusText:            u.StatusText,
+		StatusEmoji:           u.StatusEmoji,
+		StatusExpiration:      &statusExpiration,
+		Team:                  u.Team,
+	}
+}
+
+func newMqlSlackUser(runtime *plugin.Runtime, user slack.User) (plugin.Resource, error) {
+	var enterpriseUser interface{}
+
+	userProfile, err := convert.JsonToDict(newUserProfile(user.Profile))
+	if err != nil {
+		return nil, err
+	}
+
+	if user.Enterprise.ID != "" {
+		var err error
+		enterpriseUser, err = newMqlSlackEnterpriseUser(runtime, user.Enterprise)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	twoFactoryType := ""
+	if user.TwoFactorType != nil {
+		twoFactoryType = *user.TwoFactorType
+	}
+
+	return CreateResource(runtime, "slack.user", map[string]*llx.RawData{
+		"id":                llx.StringData(user.ID),
+		"teamId":            llx.StringData(user.TeamID),
+		"name":              llx.StringData(user.Name),
+		"deleted":           llx.BoolData(user.Deleted),
+		"color":             llx.StringData(user.Color),
+		"realName":          llx.StringData(user.RealName),
+		"timeZone":          llx.StringData(user.TZ),
+		"timeZoneLabel":     llx.StringData(user.TZLabel),
+		"timeZoneOffset":    llx.IntData(int64(user.TZOffset)),
+		"isBot":             llx.BoolData(user.IsBot),
+		"isAdmin":           llx.BoolData(user.IsAdmin),
+		"isOwner":           llx.BoolData(user.IsOwner),
+		"isPrimaryOwner":    llx.BoolData(user.IsPrimaryOwner),
+		"isRestricted":      llx.BoolData(user.IsRestricted),
+		"isUltraRestricted": llx.BoolData(user.IsUltraRestricted),
+		"isStranger":        llx.BoolData(user.IsStranger),
+		"isAppUser":         llx.BoolData(user.IsAppUser),
+		"isInvitedUser":     llx.BoolData(user.IsInvitedUser),
+		"has2FA":            llx.BoolData(user.Has2FA),
+		"twoFactorType":     llx.StringData(twoFactoryType),
+		"hasFiles":          llx.BoolData(user.HasFiles),
+		"presence":          llx.StringData(user.Presence),
+		"locale":            llx.StringData(user.Locale),
+		"profile":           llx.DictData(userProfile),
+		"enterpriseUser":    llx.DictData(enterpriseUser),
+	})
+}
+
+func (x *mqlSlackUser) id() (string, error) {
+	return "slack.user/" + x.TeamId.Data + "/" + x.Id.Data, nil
+}
+
+func newMqlSlackEnterpriseUser(runtime *plugin.Runtime, user slack.EnterpriseUser) (interface{}, error) {
+	return CreateResource(runtime, "slack.enterpriseUser", map[string]*llx.RawData{
+		"id":             llx.StringData(user.ID),
+		"enterpriseId":   llx.StringData(user.EnterpriseID),
+		"enterpriseName": llx.StringData(user.EnterpriseName),
+		"isAdmin":        llx.BoolData(user.IsAdmin),
+		"isOwner":        llx.BoolData(user.IsOwner),
+	})
+}
+
+func (x *mqlSlackEnterpriseUser) id() (string, error) {
+	return "slack.enterpriseUser/" + x.EnterpriseId.Data + "/" + x.Id.Data, nil
+}
+
+// init method for user
+func (s *mqlSlackUser) init(args map[string]interface{}) (map[string]interface{}, *mqlSlackUser, error) {
```

Comment:
this is not correct and won't do anything. The signature of the function must be:
```go
func initSlackUser(runtime *plugin.Runtime, args map[string]*llx.RawData) (map[string]*llx.RawData, plugin.Resource, error) {
```
then you need to rebuild the provider

Reply:
great catch 🤣